gcore/resource_gcore_waap_domain.go

```diff
@@ -0,0 +1,292 @@
+									"burst_threshold": {
+										Type:        schema.TypeInt,
+										Optional:    true,
+										Computed:    true,
+										Description: "Burst threshold for DDoS protection",
+									},
+								},
+							},
+						},
+						"api": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Computed: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"api_urls": {
+										Type:        schema.TypeList,
+										Optional:    true,
+										Description: "List of API URL patterns",
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceWaapDomainCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*Config).WaapClient
+	domainName := d.Get("name").(string)
+
+	resp, err := client.GetDomainsV1DomainsGetWithResponse(
+		context.Background(),
+		nil,
+		func(ctx context.Context, req *http.Request) error {
+			req.URL.RawQuery = "name=" + domainName
+			return nil
+		},
+	)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error listing domains: %v", err))
+	}
+
+	if resp.JSON200 != nil {
+		if resp.JSON200.Count != 0 {
+			id := fmt.Sprintf("%v", resp.JSON200.Results[0].Id)
+			d.SetId(id)
+		} else {
+			return diag.FromErr(fmt.Errorf("domain with name '%s' not found", domainName))
+		}
+	}
+
+	return resourceWaapDomainUpdate(ctx, d, m)
+}
+
+func resourceWaapDomainRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	client := m.(*Config).WaapClient
+
+	domainID, _ := strconv.Atoi(d.Get("id").(string))
+
+	// Get domain details
+	resp, err := client.GetDomainV1DomainsDomainIdGetWithResponse(
+		context.Background(),
+		domainID,
+	)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error getting domain details: %v", err))
+	}
+
+	if resp.JSON200 != nil {
+		_ = d.Set("status", string(resp.JSON200.Status))
+
+		// Get domain settings
+		settingsResp, err := client.GetDomainSettingsV1DomainsDomainIdSettingsGetWithResponse(
+			context.Background(),
+			domainID,
+		)
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("error getting domain settings: %v", err))
+		}
+
+		if settingsResp.JSON200 != nil {
+			settings := make(map[string]interface{})
+
+			if settingsResp.JSON200 != nil {
+				ddosSettings := make(map[string]interface{})
+
+				if settingsResp.JSON200.Ddos.GlobalThreshold != nil {
+					ddosSettings["global_threshold"] = *settingsResp.JSON200.Ddos.GlobalThreshold
+				}
+
+				if settingsResp.JSON200.Ddos.BurstThreshold != nil {
+					ddosSettings["burst_threshold"] = *settingsResp.JSON200.Ddos.BurstThreshold
+				}
+
+				if len(ddosSettings) > 0 {
+					settings["ddos"] = []interface{}{ddosSettings}
+				}
+			}
+
+			if settingsResp.JSON200 != nil && settingsResp.JSON200.Api.ApiUrls != nil {
```

Comment:
```suggestion
			if settingsResp.JSON200.Api.ApiUrls != nil {
```
This piece of code is inside the `settingsResp.JSON200 != nil` condition, so we don't need to check it twice.